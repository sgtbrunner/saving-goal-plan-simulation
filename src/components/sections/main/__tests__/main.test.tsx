import { screen, render, RenderResult } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import {
  CARD,
  MAIN_TITLE,
  getDefaultSavingGoal,
} from '../../../../utils/test.utils';
import Main from '../main.component';

const mockStore = configureMockStore();
const store = mockStore({ savingGoal: getDefaultSavingGoal() });

const mountComponent = (): RenderResult =>
  render(
    <Provider store={store}>
      <Main />
    </Provider>
  );

describe('Main component', () => {
  it('should match snapshot', () => {
    const { container } = mountComponent();

    expect(container).toMatchSnapshot();
  });

  it('should render main section title', () => {
    mountComponent();

    const title = screen.getByTestId(MAIN_TITLE);
    expect(title).toBeInTheDocument();
  });

  it('should render main section card', () => {
    mountComponent();

    const card = screen.getByTestId(CARD);
    expect(card).toBeInTheDocument();
  });
});
