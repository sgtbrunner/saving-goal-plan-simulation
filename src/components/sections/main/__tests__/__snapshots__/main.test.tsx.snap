// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component should match snapshot 1`] = `
<div>
  <main
    class="sc-dlnjwi iDwcvG"
    data-testid="main-component"
  >
    <p
      class="sc-hKFxyN bLgqQS"
      data-testid="main-title"
    >
      Let's plan your 
      <strong>
        saving goal
      </strong>
      .
    </p>
    <div
      class="sc-eCApnc fUBRpd"
      data-testid="card"
    >
      <div
        class="sc-jSFjdj gbIXct"
      >
        <img
          alt="Buy a house"
          src="buy-a-house.svg"
        />
        <div>
          <h1>
            Buy a house
          </h1>
          <p>
            Saving goal
          </p>
        </div>
      </div>
      <form
        class="sc-gKAaRy ljZyKU"
      >
        <div>
          <label
            class="sc-bdnxRM eJleSl"
            for="amount"
          >
            Total amount
          </label>
          <div
            class="sc-gtsrHT eNnCka"
          >
            <div
              class="sc-fujyAs hwiTAN"
            >
              <span>
                $
              </span>
            </div>
            <input
              class="sc-pNWdM eWSRPM"
              data-testid="currency-input"
              id="amount"
              inputmode="decimal"
              name="amount"
              type="text"
              value="25,000"
            />
          </div>
        </div>
        <div>
          <label
            class="sc-bdnxRM eJleSl"
            for="reachDate"
          >
            Reach goal by
          </label>
          <div
            class="sc-gtsrHT sc-kEqXSa eNnCka hbnsMv"
          >
            <button
              class="sc-jrsJWt edksVr"
              disabled=""
              id="previous-month"
              type="button"
            >
              <img
                alt="arrow-left"
                src="arrow-left.svg"
              />
            </button>
            <div
              class="sc-iqAclL ZDBoD"
            >
              <p
                class="sc-crzoAE fgafEH"
              >
                December
              </p>
              <p
                class="sc-dIsUp eVDRew"
              >
                2021
              </p>
            </div>
            <button
              class="sc-jrsJWt edksVr"
              id="next-month"
              type="button"
            >
              <img
                alt="arrow-right"
                src="arrow-right.svg"
              />
            </button>
          </div>
        </div>
      </form>
      <div
        class="sc-bqGGPW cayzAv"
      >
        <div
          class="sc-ksluID dzCQDS"
          data-testid="monthly-amount"
        >
          <h4>
            Monthly amount
          </h4>
          <p>
            $25,000
          </p>
        </div>
        <div
          class="sc-hBMUJo deFddd"
          data-testid="summary-text"
        >
          <p>
            Youâ€™re planning 
            <strong>
              1 monthly deposits
            </strong>
             
            to reach your
             
            <strong>
              $25,000
            </strong>
             goal by
             
            <strong>
              December
               
              2021
            </strong>
            .
          </p>
        </div>
      </div>
      <button
        class="sc-iCoGMd bsVzQv"
        color="primary"
        type="submit"
      >
        Confirm
      </button>
    </div>
  </main>
</div>
`;
